localization/ndt_scan_matcher/include/ndt_scan_matcher/map_update_module.hpp

```diff
@@ -44,36 +45,46 @@ class MapUpdateModule
   using PointTarget = pcl::PointXYZ;
   using NormalDistributionsTransform =
     pclomp::MultiGridNormalDistributionsTransform<PointSource, PointTarget>;
+  using NdtPtrType = std::shared_ptr<NormalDistributionsTransform>;
 
 public:
   MapUpdateModule(
     rclcpp::Node * node, std::mutex * ndt_ptr_mutex,
-    std::shared_ptr<NormalDistributionsTransform> ndt_ptr,
+    std::shared_ptr<NormalDistributionsTransform> & ndt_ptr,
```

Comment:
Isn't it possible to achieve what you want by simply using a `shared_ptr` instead of a reference to a pointer?
Sorry, but I feel like using pointer references makes the concept a bit more complex and difficult to understand.

Reply:
@SakodaShintaro I don't think it is possible.
If we just use a pointer, when we perform the assignment 
`ndt_ptr_ = secondary_ndt_ptr_`
in the map_update_module.cpp, the value of `ndt_ptr_` in ndt_scan_matcher_core.cpp **_does not_** change. That means the new scans are still aligned to the old `ndt_ptr_`, not the updated one, and that is not correct.

Reply:
I understand the idea of not copying the instance, just exchanging pointers.
Great! :+1: